blivet/populator.py

```diff
@@ -476,12 +476,10 @@ def addUdevDiskDevice(self, info):
         serial = udev.device_get_serial(info)
         bus = udev.device_get_bus(info)
 
-        # udev doesn't always provide a vendor.
-        vendor = udev.device_get_vendor(info)
-        if not vendor:
-            vendor = ""
+        vendor = util.get_sysfs_attr(sysfs_path, "device/vendor")
+        model = util.get_sysfs_attr(sysfs_path, "device/model")
```

Comment:
I did so in StorageDevice.__init__ instead so it would be covered for other callers as well.